language/compiler/ir-to-bytecode/syntax/src/syntax.rs

```diff
@@ -1209,6 +1209,19 @@ fn parse_type_var(tokens: &mut Lexer) -> Result<TypeVar, ParseError<Loc, anyhow:
     Ok(spanned(tokens.file_name(), start_loc, end_loc, type_var))
 }
 
+fn parse_type_parameter_with_phantom_decl(
+    tokens: &mut Lexer,
+) -> Result<(TypeVar, BTreeSet<Ability>, bool), ParseError<Loc, anyhow::Error>> {
+    let is_phantom = if tokens.peek() == Tok::Phantom {
```

Comment:
I would probably make `phantom` a conditional keyword. So instead of making a new token, check for an identifier where its string value is "phantom"